coc/abc.py

```diff
@@ -201,42 +208,72 @@ def _load_json_meta(cls, troop_meta, id, name, lab_to_townhall):
+                if not min_prod_unit_level:
+                    laboratory_levels = troop_meta.get("LaboratoryLevel")
+                else:
+                    # get the min th level were we can unlock by the required level of the production building
+                    min_th_level = [th for i, th in
+                                    enumerate(prod_unit["TownHallLevel"], start=1)
+                                    if i == min_prod_unit_level]
+                    # map the min th level to a lab level
+                    [first_lab_level] = [lab_level for lab_level, th_level in
+                                         lab_to_townhall.items()
+                                         if th_level in min_th_level]
+                    # the first_lab_level is the lowest possible (there are some inconsistencies with siege machines)
+                    # To handle them properly, replacing all lab_level lower than first_lab_level with first_lab_level
+                    laboratory_levels = [
+                        x if x > first_lab_level else first_lab_level
+                        for x in troop_meta.get("LaboratoryLevel")]
+            elif production_building == "Pet Shop":
+                min_prod_unit_level = troop_meta.get("LaboratoryLevel", [None, ])[0]
+                # there are some special troops, which have no BarrackLevel attribute
+
                 # get the min th level were we can unlock by the required level of the production building
-                min_th_level = [th for i, th in enumerate(prod_unit["TownHallLevel"], start=1)
+                min_th_level = [th for i, th in
+                                enumerate(prod_unit["TownHallLevel"], start=1)
                                 if i == min_prod_unit_level]
                 # map the min th level to a lab level
-                [first_lab_level] = [lab_level for lab_level, th_level in lab_to_townhall.items()
+                [first_lab_level] = [lab_level for lab_level, th_level in
+                                     lab_to_townhall.items()
                                      if th_level in min_th_level]
                 # the first_lab_level is the lowest possible (there are some inconsistencies with siege machines)
                 # To handle them properly, replacing all lab_level lower than first_lab_level with first_lab_level
-                laboratory_levels = [x if x > first_lab_level else first_lab_level
-                                     for x in troop_meta.get("LaboratoryLevel")]
+                laboratory_levels = [
+                    x if x > first_lab_level else first_lab_level
+                    for x in troop_meta.get("LaboratoryLevel")]
+            else:
+                print(1)
+                return
 
         cls.lab_level = try_enum(UnitStat, laboratory_levels)
-        cls.housing_space = _get_maybe_first(troop_meta, "HousingSpace", default=0)
+        cls.housing_space = _get_maybe_first(troop_meta, "HousingSpace",
```

Comment:
Function is starting to get to girthy. We should move the population to another function to adhere to the SOLID principle. (S - Single responsibility principle)